provider/core/src/error.rs

```diff
@@ -73,6 +73,10 @@ pub enum DataErrorKind {
     #[displaydoc("I/O error: {0:?}")]
     #[cfg(feature = "std")]
     Io(std::io::ErrorKind),
+
+    #[displaydoc("Missing source data")]
+    #[cfg(feature = "datagen")]
```

Comment:
Thought: I would've probably put this as a non-flagged enum value, but it's fine to have it flagged for now